src/ckeditor.tsx

```diff
@@ -295,45 +274,104 @@ export default class CKEditor extends React.Component {
 			initialData: this.props.config.initialData || this.props.data || ''
 		};
 	}
+
+	public static override contextType = ContextWatchdogContext;
+
+	// Properties definition.
+	public static propTypes = {
+		editor: PropTypes.func.isRequired,
+		data: PropTypes.string,
+		config: PropTypes.object,
+		watchdogConfig: PropTypes.object,
+		onChange: PropTypes.func,
+		onReady: PropTypes.func,
+		onFocus: PropTypes.func,
+		onBlur: PropTypes.func,
+		onError: PropTypes.func,
+		disabled: PropTypes.bool,
+		id: PropTypes.any,
+		// eslint-disable-next-line @typescript-eslint/no-invalid-void-type
+		onInit: ( props: Record<string, any>, propName: string ): Error | void => {
+			if ( props[ propName ] ) {
+				return new Error(
+					'The "onInit" property is not supported anymore by the CKEditor component. Use the "onReady" property instead.'
+				);
+			}
+		}
+	};
+
+	// Default values for non-required properties.
+	public static defaultProps: Partial<Props<Editor>> = {
```

Comment:
Unfortunately defaultProps disable the intellisense for required properties.

For example, with defaultProps, IDE is not showing that editor property is missing:
![image](https://user-images.githubusercontent.com/114912428/228176005-ab208dad-c91e-4354-be3f-f95216ee099a.png)

Without defaultProps:
![image](https://user-images.githubusercontent.com/114912428/228176136-ca833d5c-f082-48d4-89b1-b345eb097178.png)

I tried fixing it, but tests fail and the fix is not easy.